src/Magento/FunctionalTestingFramework/Module/MagentoActionProxies.php

```diff
@@ -0,0 +1,157 @@
+     * @return void
+     */
+    public function createEntity(
+        $key,
+        $scope,
+        $entity,
+        $dependentObjectKeys = [],
+        $overrideFields = [],
+        $storeCode = ''
+    ) {
+        if (!self::$persistHandler) {
+            self::$persistHandler = PersistedObjectHandler::getInstance();
+        }
+
+        self::$persistHandler->createEntity(
+            $key,
+            $scope,
+            $entity,
+            $dependentObjectKeys,
+            $overrideFields,
+            $storeCode
+        );
+    }
+
+    /**
+     * Retrieves and updates a previously created entity
+     *
+     * @param string $key                 StepKey of the createData action.
+     * @param string $scope
+     * @param string $updateEntity        Name of the static XML data to update the entity with.
+     * @param array  $dependentObjectKeys StepKeys of other createData actions that are required.
+     * @return void
+     */
+    public function updateEntity($key, $scope, $updateEntity, $dependentObjectKeys = [])
+    {
+        if (!self::$persistHandler) {
+            self::$persistHandler = PersistedObjectHandler::getInstance();
+        }
+
+        self::$persistHandler->updateEntity(
+            $key,
+            $scope,
+            $updateEntity,
+            $dependentObjectKeys
+        );
+    }
+
+    /**
+     * Performs GET on given entity and stores entity for use
+     *
+     * @param string  $key                 StepKey of getData action.
+     * @param string  $scope
+     * @param string  $entity              Name of XML static data to use.
+     * @param array   $dependentObjectKeys StepKeys of other createData actions that are required.
+     * @param string  $storeCode
+     * @param integer $index
+     * @return void
+     */
+    public function getEntity($key, $scope, $entity, $dependentObjectKeys = [], $storeCode = '', $index = null)
+    {
+        if (!self::$persistHandler) {
+            self::$persistHandler = PersistedObjectHandler::getInstance();
+        }
+
+        self::$persistHandler->getEntity(
+            $key,
+            $scope,
+            $entity,
+            $dependentObjectKeys,
+            $storeCode,
+            $index
+        );
+    }
+
+    /**
+     * Retrieves and deletes a previously created entity
+     *
+     * @param string $key   StepKey of the createData action.
+     * @param string $scope
+     * @return void
+     */
+    public function deleteEntity($key, $scope)
+    {
+        if (!self::$persistHandler) {
+            self::$persistHandler = PersistedObjectHandler::getInstance();
+        }
+
+        self::$persistHandler->deleteEntity($key, $scope);
+    }
+
+    /**
+     * Retrieves a field from an entity, according to key and scope given
+     *
+     * @param string $stepKey
+     * @param string $field
+     * @param string $scope
+     * @return string
+     */
+    public function retrieveEntityField($stepKey, $field, $scope)
```

Comment:
I'm not sure I understand the reason for this wrapper function, does `dry-run` throw errors if it encounters `PersistedObjectHandler::getInstance()->retrieveEntityField()` outside an acceptance tester action?

Reply:
This is needed when test trying to referencing the persisted variable. It's not really persisted and won't find it.

Reply:
![image](https://user-images.githubusercontent.com/28712019/70083230-a0e60a00-15d1-11ea-94f5-ede90cf86e4f.png)
Changing this to calls to `$I->` has an interesting side-effect (see above)
This is a test that has one `createData` and one `comment input="$createdData.Key$"`

If we absolutely *have* to include this `$I->` wrapper for any `retrieveEntityField`, we need to add an `if(step action = "retrieveEntityField")` clause to MagentoAllureAdapter so that it doesn't print out each retrieval as its own step.

This *also* has to be done with `getSecret`

Reply:
Ah, good catch! let me looking into this.

Reply:
Removed "retrieveEntityField" and "getSecret" from step display.